Fix url key of createFileOutput options for streaming

sitatec opened the pull request:
Currently, if you try to stream files you get a FileOutput instance in the following state:
```javascript
FileOutput {
  locked: false,
  [state]: 'errored',
  [supportsBYOB]: false,
  [length]: undefined
}
```
This is because a `{ data, fetch }` object is passed to `createFileOutput`, which expects a `{ url, fetch }` object.

I also noticed that FileOutput is always used for streaming, so I added the useFileOutput option to the `*stream` function and set the value of `Replicate.useFileOutput` as default.

sitatec committed:
Fix url key of createFileOutput options for streaming

sitatec committed:
Enable user to controle the use of FileOutput for streaming

u6516024155-coder reviewed (commented):
Patrycja

u6516024155-coder reviewed (commented):
Super aplikacja😘

rovo89 commented:
I have just spend at least two hours trying to find out what I'm doing wrong, trying to write the output to a file. Besides the fact that `ServerSentEvent.data` is a string and needs to be converted to a `ReadableStream` or `FileOutput` first, the crucial point was the wrong call of `createFileOutput()` that is fixed with this PR.

@zeke Could you merge at least this part of the suggested change? Not sure about the other half and the types, but for the wrong call there is no workaround. Thanks!

aron committed:
Detect file streams by looking at URL path prefix

The file streaming API is hosted at
https://stream.replicate.com/v1/files so we can use this prefix to
determine that the response will contain file entities. This is far
cleaner than relying on the event contents.

aron committed:
Add tests for the file streaming interface

aron committed:
Format biome.json

aron commented:
I will cut a new release shortly.

aron commented:
This has now been released as 1.4.0
